### ISSUE TITLE
CasADi 3.6.6 Crashes on macOS with EXC_BAD_ACCESS (SIGKILL)

### ISSUE THREAD

**Issue by aferrarelli** [casadi/casadi#3833]
### Description:
I'm experiencing a critical issue when using CasADi version 3.6.6 on my MacBook Pro M3 running macOS 14.1. The script that worked perfectly with CasADi 3.6.5 now crashes immediately when running with version 3.6.6. The error reported by macOS is:

```
Exception Type:        EXC_BAD_ACCESS (SIGKILL (Code Signature Invalid))
```

The crash occurs consistently when attempting to execute any script using CasADi 3.6.6. Reverting to version 3.6.5 resolves the issue, and the same script runs without any problems.

**System Details:**
- MacBook Pro M3
- macOS 14.1
- Python 3.12.5
- CasADi 3.6.6 (causing the crash)
- CasADi 3.6.5 (works fine)

Thanks

**Comment by faultdiagnosistoolbox** [comment]
Same here, tried example race_car.py on 3.6.6 and it crashes. Reverting to 3.6.5 runs without a problem.

**Comment by jgillis** [comment]
```
$ codesign --verify --deep --verbose /Users/m1/miniconda3/lib/python3.12/site-packages/casadi/libipopt.3.dylib
/Users/m1/miniconda3/lib/python3.12/site-packages/casadi/libipopt.3.dylib: invalid signature (code or signature have been modified)
In architecture: arm64
```
```
$ python casadi/docs/examples/python/race_car.py || log show --last 2s --debug 
2024-09-28 21:37:51.829845+0200 0x167732   Default     0x0                  0      0    kernel: CODE SIGNING: cs_invalid_page(0x117438000): p=46800[python3.12] final status 0x23000200, denying page sending SIGKILL
2024-09-28 21:37:51.829857+0200 0x167732   Default     0x0                  0      0    kernel: CODE SIGNING: process 46800[python3.12]: rejecting invalid page at address 0x117438000 from offset 0x0 in file "/Users/m1/miniconda3/lib/python3.12/site-packages/casadi/libipopt.3.dylib" (cs_mtime:1727550431.514931197 == mtime:1727550431.514931197) (signed:1 validated:1 tainted:1 nx:0 wpmapped:0 dirty:0 depth:0)
2024-09-28 21:37:51.829924+0200 0x167732   Default     0x0                  0      0    kernel: python3.12[46800] Corpse allowed 1 of 5
```

**Comment by jgillis** [comment]
Fixed in #3788
